File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equipable/ArcaneBattery.java
package org.rexcellentgames.burningknight.entity.item.accessory.equipable;

import org.rexcellentgames.burningknight.assets.Locale;
import org.rexcellentgames.burningknight.entity.creature.player.Player;

public class ArcaneBattery extends Equipable {
	{
		name = Locale.get("arcane_battery");
		description = Locale.get("arcane_battery_desc");
		sprite = "item-lower_health_more_mana_regen";
	}

	@Override
	public void onEquip() {
		super.onEquip();

		if (this.owner instanceof Player) {
			((Player) this.owner).moreManaRegenWhenLow = true;
		}
	}

	@Override
	public void onUnequip() {
		super.onUnequip();

		if (this.owner instanceof Player) {
			((Player) this.owner).moreManaRegenWhenLow = false;
		}
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equipable/ArcaneBattery.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equipable

import org.rexcellentgames.burningknight.assets.Locale

class ArcaneBattery : Equipable() {
	init {
		name = Locale.get("arcane_battery")
		description = Locale.get("arcane_battery_desc")
		sprite = "item-lower_health_more_mana_regen"
	}

	override fun canBeUpgraded(): Boolean {
		return false
	}

	override fun onEquip() {
		super.onEquip()
		this.owner.moreManaRegenWhenLow = true
	}

	override fun onUnequip() {
		super.onUnequip()
		this.owner.moreManaRegenWhenLow = false
	}
}